### PR TITLE
task completed

### PR DESCRIPTION
**Pull request by voronvasia**

### CODE REVIEW

File: app/test_main.py [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+        "1 extra human year.",
+        "first 15 cat and dog years give 1 human year, next 9 cat "
+        "and dog years give 1 more human "
+        "year every 4 next cat years (and 5 next dog years) give "
+        "1 extra human year.",
+
+    ]
+)
+def test_correspondence_of_human_years_to_animals(
+        cat_age: int, dog_age: int, human_age: int
```
Comment (Dimosphen1) [inline app/test_main.py:36]: Place each argument on a new line to improve the readability.

File: app/test_main.py [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+
+@pytest.mark.parametrize(
+    "cat_age,dog_age,human_age",
+    [
+        (0, 0, [0, 0]),
+        (14, 14, [0, 0]),
+        (15, 15, [1, 1]),
+        (24, 24, [2, 2]),
+        (28, 28, [3, 2]),
+        (100, 100, [21, 17])
```
Comment (Dimosphen1) [inline app/test_main.py:14]: Add test cases to check input when `cat_age` and `dog_age` have different values so that they are not dependent. Besides that, add test cases to test against negative numbers and incorrect types of data that raise the correct exception. you could separate the logic to test exceptions to a distinct function. Follow the checklist to make sure that all required cases are covered.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

**Review by Dimosphen1 — APPROVED** [review]
LGTM, well done!
